### PR TITLE
[WIP] Add libdatachannel

### PR DESCRIPTION
**Pull request by DDRBoxman**
### Checklist:

- [ ] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [ ] The code has been tested.
- [ ] All commit messages are properly formatted and commits squashed where appropriate.
- [ ] I have included updates to all appropriate documentation.

### CODE REVIEW

File: deps.macos/90-libdatachannel.zsh [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  log_info "Config (%F{3}${target}%f)"
+
+  args=(
+    ${cmake_flags}
+    -DUSE_MBEDTLS=1
+    -DNO_WEBSOCKET=1
+    -DNO_TESTS=1
+    -DNO_EXAMPLES=1
+    -DPYTHON_EXECUTABLE=$(brew --prefix)/opt/python3/bin/python3
+    -DPYTHON_LIBRARY=$(brew --prefix)/opt/python3/lib/
```
Comment (tytan652) [inline deps.macos/90-libdatachannel.zsh:37]: You are building Mbeb TLS submodule inside of libdatachannel and not using the "system" one.

```suggestion
    -DUSE_SYSTEM_MBEDTLS=ON
```

File: requirements.txt
Comment (tytan652) [inline requirements.txt:1]: Not needed if we skip Mbed TLS submodule.

File: utils.zsh/20-host.zsh [hunk 2]
```diff
@@ -27,6 +27,8 @@ _setup_macos() {
 
     log_status "Installing required build dependencies..."
     brew ${_brew_params}
+
+    python3 -m pip install -r "${project_root}"/requirements.txt
```
Comment (tytan652) [inline utils.zsh/20-host.zsh:31]: Not needed if we skip Mbed TLS submodule.

**Review by tytan652 — CHANGES_REQUESTED** [review]
Please ignore this one.

File: deps.macos/90-libdatachannel.zsh [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+  log_info "Config (%F{3}${target}%f)"
+
+  args=(
+    ${cmake_flags}
+    -DUSE_MBEDTLS=1
+    -DNO_WEBSOCKET=1
+    -DNO_TESTS=1
+    -DNO_EXAMPLES=1
+    -DUSE_SYSTEM_MBEDTLS=1
```
Comment (tytan652) [inline deps.macos/90-libdatachannel.zsh:36]:
```suggestion
```

File: deps.windows/70-libdatachannel.ps1 [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    Log-Information "Configure (${Target})"
+    Set-Location $Path
+
+    $Options = @(
+        $CmakeOptions
+        '-DUSE_MBEDTLS=1'
+        '-DNO_WEBSOCKET=1'
+        '-DNO_TESTS=1'
+        '-DNO_EXAMPLES=1'
+        '-DUSE_SYSTEM_MBEDTLS=1'
```
Comment (tytan652) [inline deps.windows/70-libdatachannel.ps1:41]:
```suggestion
```

**Review by tytan652 — CHANGES_REQUESTED** [review]
`USE_SYSTEM_MBEDTLS` no longer exist since the submodule got dropped recently.

**Comment by RytoEX** [comment]
> We still need a solution to deal with the fact that on windows we now have a mbedtls from mingw and a mbedtls compiled with msvc.

We used to do this before the obs-deps rewrite. Check back in the git history between November 2021 and May 2022. I think we just copied the native build over top of the MinGW build, because the MinGW build was only used for libsrt or something like that.

**Pull request closed without merging**